### PR TITLE
Introduce transfer config command

### PR DESCRIPTION
**Pull request by yahavi**
- [x] All [tests](https://github.com/jfrog/jfrog-cli-core#tests) passed. If this feature is not already covered by the tests, I added new tests.
- [x] All [static analysis checks](https://github.com/jfrog/jfrog-cli-core/actions/workflows/analysis.yml) passed.
- [x] This pull request is on the dev branch.
- [x] I used gofmt for formatting the code before submitting the pull request.
-----

Depend on https://github.com/jfrog/jfrog-client-go/pull/570, https://github.com/jfrog/config-import/pull/2
Required for https://github.com/jfrog/jfrog-cli/pull/1546

### CODE REVIEW

File: artifactory/commands/transferconfig/utils.go [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+		if err != nil {
+			return nil, err
+		}
+		if _, retErr = fileWriter.Write(fileContent); retErr != nil {
+			return
+		}
+	}
+	fileWriter, err := writer.Create("artifactory.config.xml")
+	if err != nil {
+		return buffer, err
```
Comment (eyalbe4) [inline artifactory/commands/transferconfig/utils.go:61]: Missing CheckError here.

File: artifactory/commands/transferconfig/utils.go [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+		if err != nil {
+			if os.IsNotExist(err) && strings.HasSuffix(neededFile, "url.signing.key") {
+				log.Info("url.signing.key file is missing in the source Artifactory server. Skipping...")
+				continue
+			}
+			return nil, err
+		}
+		fileWriter, err := writer.Create(neededFile)
+		if err != nil {
+			return nil, err
```
Comment (eyalbe4) [inline artifactory/commands/transferconfig/utils.go:53]: Missing CheckError here.

File: artifactory/commands/transferconfig/utils.go [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+		if retErr == nil {
+			retErr = closeErr
+		}
+	}()
+
+	for _, neededFile := range neededFiles {
+		neededFilePath := filepath.Join(exportPath, neededFile)
+		log.Debug("Archiving " + neededFile)
+		fileContent, err := os.ReadFile(neededFilePath)
+		if err != nil {
```
Comment (eyalbe4) [inline artifactory/commands/transferconfig/utils.go:44]: Missing CheckError here.

File: artifactory/commands/transferconfig/utils.go [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+func archiveConfig(exportPath string, configXml string) (buffer *bytes.Buffer, retErr error) {
+	buffer = &bytes.Buffer{}
+	writer := zip.NewWriter(buffer)
+	writer.RegisterCompressor(zip.Deflate, func(out io.Writer) (io.WriteCloser, error) {
+		return flate.NewWriter(out, flate.BestCompression)
+	})
+	defer func() {
+		closeErr := writer.Close()
+		if retErr == nil {
+			retErr = closeErr
```
Comment (eyalbe4) [inline artifactory/commands/transferconfig/utils.go:36]: Missing CheckError here.

File: artifactory/commands/transferconfig/utils.go [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+		}
+		if _, retErr = fileWriter.Write(fileContent); retErr != nil {
+			return
+		}
+	}
+	fileWriter, err := writer.Create("artifactory.config.xml")
+	if err != nil {
+		return buffer, err
+	}
+	_, retErr = fileWriter.Write([]byte(configXml))
```
Comment (eyalbe4) [inline artifactory/commands/transferconfig/utils.go:63]: Missing CheckError here.

**Review by eyalbe4 — CHANGES_REQUESTED** [review]

File: artifactory/commands/transferconfig/transferconfig.go [hunk 6]
```diff
@@ -0,0 +1,307 @@
...
+		return err
+	}
+
+	// Import the archive to the target Artifactory
+	return tcc.importToTargetArtifactory(targetServiceManager, archiveConfig)
+}
+
+// Make sure source and target Artifactory URLs are different.
+// Make sure the target Artifactory is empty, by counting the number of the repositories. If it is bigger than 1, return an error.
+func (tcc *TransferConfigCommand) verifyArtifactoryServers(targetServicesManager artifactory.ArtifactoryServicesManager, sourceArtifactoryVersion string) error {
```
[the remark this answers is not in the document]
Comment (yahavi) [reply artifactory/commands/transferconfig/transferconfig.go]: Will be implemented in a follow-up PR + https://github.com/jfrog/config-import/pull/3

File: artifactory/commands/transferconfig/transferconfig.go [hunk 7]
```diff
@@ -0,0 +1,307 @@
...
+	if tcc.force {
+		return nil
+	}
+	log.Info("Verifying target server is empty...")
+	filterParams := services.RepositoriesFilterParams{RepoType: utils.LOCAL.String()}
+	localRepositories, err := targetServicesManager.GetAllRepositoriesFiltered(filterParams)
+	if err != nil {
+		return err
+	}
+	if len(*localRepositories) > 1 {
```
[the remark this answers is not in the document]
Comment (yahavi) [reply artifactory/commands/transferconfig/transferconfig.go]: Changed to "number of users > 2"

**Pull request merged**